### PR TITLE
Add swipejs@2.2.4 w/ git auto-update

### LINKED ISSUE #10016: [Request] Add lyfeyaj/swipe
**Library name:** swipe
**Git repository url:** https://github.com/lyfeyaj/swipe
**npm package url(optional):** https://www.npmjs.com/package/swipejs
**License(s):** MIT
**Official homepage:** http://lyfeyaj.github.io/swipe/
**Wanna say something? Leave message here:**

This is a modern fork of [thebird/swipe](https://github.com/thebird/Swipe) which is already hosted on CDNJS. The problem is that the library hasn’t been updated for about 2-3 years. So this library should be there as well since it is being actively maintained.

### PR DESCRIPTION
**Pull request by sufuf3**
close #10016, cc @lyfeyaj
@kennynaoh please help me review this PR, thank you.

Pull request for issue: #10016 
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/lyfeyaj/swipe
 * Official website (optional, not the repository): http://lyfeyaj.github.io/swipe
 * NPM package url (optional): https://www.npmjs.com/package/swipejs
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 13
   - Count of stars: 228
   - Count of forks: 46
 * NPM download stats (optional):
   - Downloads in the last day: 77
   - Downloads in the last week: 379
   - Downloads in the last month: 1553

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [ ] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by sufuf3** [comment]
@kennynaoh I modified it, please help me review this PR again, thank you.

**Comment by kennynaoh** [comment]
@sufuf3 Why do you delete some files of old versions in the first commit?

**Comment by sufuf3** [comment]
@kennynaoh I modified it, please help me review this PR again, thank you.

**Comment by PeterDaveHello** [comment]
The origin homepage redirects to another repo https://github.com/thebird/swipe/, confirmed?

**Comment by pvnr0082t** [comment]
@sufuf3 #10016 seems like not request to update lib. It seems like to add a new lib.

**Comment by sufuf3** [comment]
@PeterDaveHello @pvnr0082t  
With the note of lib ([lyfeyaj/swipe](https://github.com/lyfeyaj/swipe#note))
It say "This repo is a continuation of the dead Swipe project. Our mission is to keep Swipe alive and moving forward, with maintenance fixes and new features. Pull Requests are welcome!"
And the 1.0 and 2.0.0 version of this repo, is same form [origin repo](https://github.com/thebird/Swipe)

So I think it need update.

**Comment by PeterDaveHello** [comment]
Does the origin author agree with this situation? If you don't know, maybe you should try to confirm that, or how can you take somebody's work with its origin place without any agreement?

**Comment by PeterDaveHello** [comment]
Any updates here? Thanks!

**Comment by sufuf3** [comment]
The the note of lib ([lyfeyaj/swipe](https://github.com/lyfeyaj/swipe#note))
It says "This repo is a continuation of the dead Swipe project. Our mission is to keep Swipe alive and moving forward, with maintenance fixes and new features. Pull Requests are welcome!"
And the 1.0 and 2.0.0 version of this repo is same form [origin repo](https://github.com/thebird/Swipe)

I try to discuss with the origin author that did he agree with this situation.
But until now, there is no response from the origin author.
Maybe it's the dead Swipe project.
And I think we should update from new repository, maybe.

**Comment by PeterDaveHello** [comment]
If there is no response, just add it as a new lib, do not mix them up.

**Comment by sufuf3** [comment]
@cdnjs/intern3 Please help me review this pull request, thank you.

File: ajax/libs/swipejs/package.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/lyfeyaj/swipe.git"
+  },
+  "keywords": [
+    "swipe",
+    "slider",
+    "touch"
+  ],
+  "homepage": "http://lyfeyaj.github.io/swipe/",
```
Comment (extend1994) [inline ajax/libs/swipejs/package.json]: Use https

File: ajax/libs/swipejs/package.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+          "swipe*"
+        ]
+      },
+      {
+        "basePath": "build",
+        "files": [
+          "**/*"
+        ]
+      }
+    ]
```
Comment (extend1994) [inline ajax/libs/swipejs/package.json]: I think we also need `style.css` whose `basePath` field is `""`.
Ref: https://github.com/lyfeyaj/swipe#style
> Swipe requires the following styles to be added to your stylesheet: ...

It's the same as `style.css`.

**Comment by sufuf3** [comment]
@extend1994 I added the style.css and modified the homepage.
Please help me review this PR, thanks.

File: ajax/libs/swipejs/package.json [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  "homepage": "https://lyfeyaj.github.io/swipe/",
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/lyfeyaj/swipe.git",
+    "fileMap": [
+      {
+        "basePath": "",
+        "files": [
+          "swipe*",
+          "style*"
```
Comment (extend1994) [inline ajax/libs/swipejs/package.json]: LGTM
But I am not sure the rule `style*` is suitable or not.

Comment (PeterDaveHello) [inline ajax/libs/swipejs/package.json]: should be more precise.

Comment (sufuf3) [inline ajax/libs/swipejs/package.json]: How about use `style.css`?
Is it OK?

Comment (PeterDaveHello) [inline ajax/libs/swipejs/package.json]: fine

**Comment by sufuf3** [comment]
@cdnjs/intern3 I modified form `style*` to `style.css`.
Please help me review this PR, thanks.

**Comment by sufuf3** [comment]
The root one is correct.
https://github.com/lyfeyaj/swipe/issues/63#issuecomment-291743535

**Comment by PeterDaveHello** [comment]
Then you should make sure it'll grab from root.

**Comment by sufuf3** [comment]
It is
```
       "basePath": "",
        "files": [
          "swipe*",
          "style.css"
        ]
```

**Pull request merged**